tests/features/support/steps.js

```diff
@@ -51,38 +110,158 @@ When(/I click the (button|link) "([^"]+)"/, async (elemType, phrase) => {
 
   if (elem) {
     await Promise.all([
-      elem.click(),
+      elem.click(),  // TODO: change to `clickOrTap`
       scope.page.waitForNavigation({waitUntil: 'domcontentloaded'})
-  ]);
+    ]);
   } else {
     if (process.env.DEBUG) {
-      await scope.page.screenshot({ path: './error.jpg', type: 'jpeg' });
+      await scope.page.screenshot({ path: './error.jpg', type: 'jpeg', fullPage: true });
     }
     throw `No ${elemType} with text ${phrase} exists.`;
   }
 });
 
 When('I click the defined text link {string}', async (phrase) => {
+  // Should we wait for navigation here?
   const [link] = await scope.page.$x(`//a[contains(text(), "${phrase}")]`);
   if (link) {
-    await link.click();
+    await link.click();  // TODO: change to `clickOrTap`
   } else {
     if (process.env.DEBUG) {
-      await scope.page.screenshot({ path: './error.jpg', type: 'jpeg' });
+      await scope.page.screenshot({ path: './error.jpg', type: 'jpeg', fullPage: true });
     }
     throw `No link with text ${phrase} exists.`;
   }
 });
 
+Then('I should see the link {string}', async (linkText) => {
+  let [link] = await scope.page.$x(`//a[contains(text(), "${linkText}")]`);
+  expect(link).to.exist;
+});
+
+// TODO: Research how da handles invalid symbols
+Then('the question id should be {string}', async (question_class) => {
+  /* Looks for the question id as it's written in the .yml.
```

Comment:
Is this looking for a question ID or question class? The sentence says ID but the `waitFor` function below is looking for a class.

Reply:
This refers to the docassemble `question` and `id`:
```
id: user name
question: |
  What is your name?
# And so on
```

These cucumber scripts are meant to be written mostly by people that are familiar with docassemble. Many of the folks I've been working with don't know html, javascript, how to examine the DOM, etc.

Reply:
Something that does need to be a future issue is how to sanitize the id text in the same way that docassemble does it so that special characters can be handled. I think I have a note to that effect.

Reply:
If we do add looking for actual html ids, my plan is to use the text, 'The element with the id {string} is present' or something.